dlt/common/configuration/specs/azure_credentials.py

```diff
@@ -0,0 +1,61 @@
+from typing import Optional, Dict, Any
+
+from dlt.common import pendulum
+from dlt.common.exceptions import MissingDependencyException
+from dlt.common.typing import TSecretStrValue
+from dlt.common.configuration.specs import CredentialsConfiguration, CredentialsWithDefault, configspec
+from dlt.common.configuration.specs.exceptions import InvalidBoto3Session
+from dlt import version
+
+import fsspec
+
+
+@configspec
+class AzureCredentialsWithoutDefaults(CredentialsConfiguration):
+    """Credentials for azure blob storage, compatible with adlfs"""
+
+    azure_storage_account_name: str = None
+    azure_storage_account_key: Optional[TSecretStrValue] = None
+    azure_storage_sas_token: TSecretStrValue = None
+
+    def to_adlfs_credentials(self) -> Dict[str, Any]:
+        """Return a dict that can be passed as kwargs to adlfs"""
+        return dict(
+            account_name=self.azure_storage_account_name,
+            account_key=self.azure_storage_account_key,
+            sas_token=self.azure_storage_sas_token,
+        )
+
+    def create_sas_token(self) -> None:
+        from azure.storage.blob import generate_account_sas, ResourceTypes
```

Comment:
I hope this dependency is a part of adlfs...

Reply:
Yes, it's included with the azure sdks

```
poetry show adlfs
 name         : adlfs                                           
 version      : 2023.8.0                                        
 description  : Access Azure Datalake Gen1 with fsspec and dask 

dependencies
 - aiohttp >=3.7.0
 - azure-core >=1.23.1,<2.0.0
 - azure-datalake-store >=0.0.46,<0.1
 - azure-identity *
 - azure-storage-blob >=12.12.0
 - fsspec >=2021.10.1
 ```